bindings/go/src/fdb/transact.go

```diff
@@ -0,0 +1,116 @@
+/*
+ * database.go
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// FoundationDB Go API
+package fdb
+
+func Transact(db Database, f func(t Transaction) error) error {
+	_, err := db.Transact(func(t Transaction) (interface{}, error) {
+		return nil, f(t)
+	})
+	return err
+}
+
+func Transact1[T any](db Database, f func(t Transaction) (T, error)) (T, error) {
```

Comment:
Could you provide some examples in which cases those different `Transact1`, `Transact2` and `Transact3` are needed?